crates/nu-cli/src/commands/to_html.rs

```diff
@@ -60,183 +159,135 @@ impl WholeStreamCommand for ToHTML {
 
-    hm
-}
+    // If asset doesn't work, make sure to return the default theme
+    let asset = match asset {
+        Ok(a) => a,
+        _ => HtmlThemes::default(),
+    };
 
-fn get_default_theme(is_dark: bool) -> HashMap<&'static str, String> {
-    let mut hm: HashMap<&str, String> = HashMap::new();
+    // Find the theme by theme name
+    let th = asset
+        .themes
+        .iter()
+        .find(|&n| n.name.to_lowercase() == *theme_name.to_lowercase().as_str()); // case insensitive search
+
+    // If no theme is found by the name provided, ensure we return the default theme
+    let default_theme = HtmlTheme::default();
+    let th = match th {
+        Some(t) => t,
+        None => &default_theme,
+    };
 
-    // This theme has different colors for dark and light
-    // so we can't just swap the background colors.
-    if is_dark {
-        hm.insert("bold_black", "black".to_string());
-        hm.insert("bold_red", "red".to_string());
-        hm.insert("bold_green", "green".to_string());
-        hm.insert("bold_yellow", "yellow".to_string());
-        hm.insert("bold_blue", "blue".to_string());
-        hm.insert("bold_magenta", "magenta".to_string());
-        hm.insert("bold_cyan", "cyan".to_string());
-        hm.insert("bold_white", "white".to_string());
-
-        hm.insert("black", "black".to_string());
-        hm.insert("red", "red".to_string());
-        hm.insert("green", "green".to_string());
-        hm.insert("yellow", "yellow".to_string());
-        hm.insert("blue", "blue".to_string());
-        hm.insert("magenta", "magenta".to_string());
-        hm.insert("cyan", "cyan".to_string());
-        hm.insert("white", "white".to_string());
-
-        hm.insert("background", "black".to_string());
-        hm.insert("foreground", "white".to_string());
-    } else {
-        hm.insert("bold_black", "black".to_string());
-        hm.insert("bold_red", "red".to_string());
-        hm.insert("bold_green", "green".to_string());
-        hm.insert("bold_yellow", "#717100".to_string());
-        hm.insert("bold_blue", "blue".to_string());
-        hm.insert("bold_magenta", "#c800c8".to_string());
-        hm.insert("bold_cyan", "#037979".to_string());
-        hm.insert("bold_white", "white".to_string());
-
-        hm.insert("black", "black".to_string());
-        hm.insert("red", "red".to_string());
-        hm.insert("green", "green".to_string());
-        hm.insert("yellow", "#717100".to_string());
-        hm.insert("blue", "blue".to_string());
-        hm.insert("magenta", "#c800c8".to_string());
-        hm.insert("cyan", "#037979".to_string());
-        hm.insert("white", "white".to_string());
-
-        hm.insert("background", "white".to_string());
-        hm.insert("foreground", "black".to_string());
+    if th.name.to_lowercase().eq(&"campbell".to_string()) // this just means no theme was passed in
+        && theme.is_some()
+    // this means there was a theme passed in
+    {
+        println!(
```

Comment:
This should probably be a ShellError, I think?

Reply:
ok. I may need help with that. I'll dig around in the source to see if I can find an example of what that looks like.

Reply:
Now that I think about this, I was printing a warning here, notifying the user that their theme name couldn't be found and that I was using the default theme. If I use ShellError, it appears that I have to fail versus give a warning. I'm not sure which we prefer to do. Any suggestions?

Reply:
We probably generally want some guidelines for when we give a warning vs an error. For this case, I think it'd be okay to error for now. In the future, we could do a pass over where we error and see if some of those could become warnings.